okpt/io/args.py

```diff
@@ -27,90 +27,166 @@
-    _add_config_path_arg(parser_test, 'config_path')
-    _add_output_path_arg(parser_test, 'output_path')
-
+def _add_output(parser, name, **kwargs):
+    """"Add output file path argument."""
+    opts = {
+        'type': _write_type,
+        'help': 'Path of output file.',
+        'metavar': 'output_path',
+        **kwargs,
+    }
+    parser.add_argument(name, **opts)
 
-def _add_plot_subcommand(subparsers):
-    parser_plot = subparsers.add_parser('plot')
-    _add_config_path_arg(parser_plot, 'config_path')
-    _add_results_paths_arg(parser_plot, 'results_paths')
 
+def _add_metadata(parser, name, **kwargs):
+    opts = {
+        'action': 'store_true',
+        **kwargs,
+    }
+    parser.add_argument(name, **opts)
 
-def _add_compare_subcommand(subparsers):
-    parser_compare = subparsers.add_parser('compare')
-    _add_config_path_arg(parser_compare, 'config_path')
-    _add_output_path_arg(parser_compare, 'output_path')
-    _add_results_paths_arg(parser_compare, 'results_paths')
 
+def _add_test_cmd(subparsers):
+    test_parser = subparsers.add_parser('test')
+    _add_config(test_parser, 'config')
+    _add_output(test_parser, 'output')
 
-_parser = argparse.ArgumentParser(
-    description=
-    'Run performance tests against the OpenSearch plugin and various ANN libaries.'
-)
 
+def _add_diff_cmd(subparsers):
+    diff_parser = subparsers.add_parser('diff')
+    _add_metadata(diff_parser, '--metadata')
+    _add_result(diff_parser,
+                'base_result',
+                help='Base test result.',
+                metavar='base_result')
+    _add_result(diff_parser,
+                'changed_result',
+                help='Changed test result.',
+                metavar='changed_result')
+    _add_output(diff_parser, '--output', default=sys.stdout)
 
-def define_args():
-    """Define tool commands."""
-    _parser.add_argument(
-        '--log',
-        type=str,
-        choices=['debug', 'info', 'warning', 'error', 'critical'],
-        default='info')
-    subparsers = _parser.add_subparsers(title='commands',
-                                        dest='command',
-                                        help='sub-command help')
-    subparsers.required = True
 
-    # add subcommands
-    _add_test_subcommand(subparsers)
-    _add_plot_subcommand(subparsers)
-    _add_compare_subcommand(subparsers)
+def _add_plot_cmd(subparsers):
+    plot_parser = subparsers.add_parser('plot')
+    _add_results(plot_parser, 'results')
+    _add_output(plot_parser, 'output')
 
 
 @dataclass
-class ToolArgs:
+class TestArgs:
     log: str
     command: str
     config: TextIOWrapper
     output: TextIOWrapper
 
 
-def get_args() -> ToolArgs:
-    """Parses and returns the command line args.
+@dataclass
+class DiffArgs:
+    log: str
+    command: str
+    metadata: bool
+    base_result: TextIOWrapper
+    changed_result: TextIOWrapper
+    output: TextIOWrapper
+
+
+@dataclass
+class PlotArgs:
```

Comment:
Why are you adding plot in this PR?

Reply:
It seemed to go hand in hand with the `args.py` refactor, as I felt it should've been added into the configuration PR from earlier. Should I remove it?

Reply:
Yes, I think plotting is outside scope of this PR.

Reply:
Plot is removed